src/components/PostDetails.tsx

```diff
@@ -65,28 +34,14 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
         postId: post.id,
       });
 
-      setComments(
-        currentComments => [...currentComments, newComment],
-      );
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
+      dispatch(setComments([...comments, newComment]));
     } catch (error) {
-      // we show an error message in case of any error
       setError(true);
     }
   };
 
   const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    setComments(
-      currentComments => currentComments.filter(
-        comment => comment.id !== commentId,
-      ),
-    );
+    dispatch(setComments(comments.filter(comment => comment.id !== commentId)));
```

Comment:
Same here. 
If we work with redux let's use all it's power. Move this logic inside commentsSlice and create a new action with will only delete comments by id
```suggestion
    dispatch(deleteComment(commentId));
```